torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1544,18 +1544,40 @@ Example::
 
   module.def(
       "_compute_bucket_assignment_by_size",
-      &::c10d::compute_bucket_assignment_by_size,
+      [](const std::vector<at::Tensor>& tensors,
+            const std::vector<size_t>& bucket_size_limits,
+            const std::vector<bool>& expect_sparse_gradient,
+            const std::vector<int64_t>& tensor_indices,
+            const c10::optional<std::weak_ptr<::c10d::Logger>>& logger) {
+             if (logger.has_value()) {
```

Comment:
Great job figuring out the pybind stuff so quickly! 

You've modified `c10d::compute_bucket_by_size` to already take an optional logger, so I don't think we need the if/else branch here. If `logger` doesn't have the value set it will be appropriately handled in the reducer so just pass `logger` regardless here?

Reply:
Nevermind, forgot that we need this to be weak_ptr. The branching based on `has_value` looks fine to me. 

Can we make this pybind similar to `set_logger` pybind, but it will have c10::optional? i.e. pass in 

`const c10::optional<std::shared_ptr<::c10d::Logger>> logger`.